src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -295,7 +398,14 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         for (uint8_t i = 0; i < TableColorNames.size(); ++i)
         {
-            _CurrentColorTable.GetAt(i).Color(colorScheme.Table()[i]);
+            if (i < ColorTableDivider)
+            {
+                _CurrentNonBrightColorTable.GetAt(i).Color(colorScheme.Table()[i]);
+            }
+            else
+            {
+                _CurrentBrightColorTable.GetAt(i - 8).Color(colorScheme.Table()[i]);
+            }
```

Comment:
```suggestion
            if (i < ColorTableDivider)
            {
                _CurrentNonBrightColorTable.GetAt(i).Color(colorScheme.Table()[i]);
            }
            else
            {
                _CurrentBrightColorTable.GetAt(i - ColorTableDivider).Color(colorScheme.Table()[i]);
            }
```
@htcfreek This code is correct. I'm mapping...
- `_CurrentNonBrightColorTable[0]` --> `colorScheme.Table()[0]` through `_CurrentNonBrightColorTable[7]` --> `colorScheme.Table()[7]`
- `_CurrentBrightColorTable[0]` --> `colorScheme.Table()[8]` through `_CurrentBrightColorTable[7]` --> `colorScheme.Table()[15]`

The suggestion I added above is for clarity.

#8997 was...
- adding the `ColorTableDivider` instead of subtracting it
- populating both lists with 16 colors (instead of half and half)
- probably getting an index-out-of-bounds error on line 337 when trying to access `colorScheme.Table()[indexBrightColor]` when `indexBrightColor = 16`

Reply:
How many items have `_CurrentNonBrightColorTable` and `_CurrentBrightColorTable`? I thought 7 8 items (0-7).

Your index (i) goes from 0-15 if I saw it correct. Your condition is `< TableColorNames.Size()` and `< TableColorNames.Size()` has 16 items.

Reply:
Correct, but the condition on line 401
```c++
if(i < ColorTableDivider)
```
makes it so that the items are conditionally updated. So the first 8 affect `_CurrentNonBrightColorTable` whereas the remaining 8 affect `_CurrentBrightColorTable`.

Reply:
But for the second 8 colors `i` is `8` to `15` and `_CurrentBrightColorTable` has element `0` to `7`. Correct?

So we get `_CurrentBrightColorTable[8]` to `_CurrentBrightColorTable[16]`. Correct?